integration/restart_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/gravitational/teleport/lib"
+	"github.com/gravitational/teleport/lib/auth/testauthority"
+	"github.com/gravitational/teleport/lib/service"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+// TestLostConnectionToAuthCausesReload tests that a lost connection to the auth server
+// will eventually restart a node
+func TestLostConnectionToAuthCausesReload(t *testing.T) {
+	// Because testing that the node does a full restart is a bit tricky when
+	// running a cluster from inside a test runner (i.e. we don't want to
+	// SIGTERM the test runner), we will watch for the node emitting a
+	// `TeleportReload` even instead. In a proper Teleport instance, this
+	// event would be picked up at the Supervisor level and would eventually
+	// cause the instance to gracefully restart.
+
+	require := require.New(t)
+	log := log.StandardLogger()
+
+	log.Info(">>> Entering Test")
+
+	// InsecureDevMode needed for SSH connections
+	// TODO(tcsc): surface this as per-server config (see also issue #8913)
+	lib.SetInsecureDevMode(true)
+	defer lib.SetInsecureDevMode(false)
+
+	// GIVEN  a cluster with a running auth+proxy instance....
+	log.Info(">>> Creating cluster")
+	keygen := testauthority.New()
+	privateKey, publicKey, err := keygen.GenerateKeyPair("")
+	require.NoError(err)
+	auth := NewInstance(InstanceConfig{
+		ClusterName: "test-tunnel-collapse",
+		HostID:      "auth",
+		Priv:        privateKey,
+		Pub:         publicKey,
+		Ports:       standardPortSetup(),
+		log:         log,
+	})
+
+	log.Info(">>> Creating auth-proxy...")
+	authConf := service.MakeDefaultConfig()
+	authConf.Hostname = Host
+	authConf.Auth.Enabled = true
+	authConf.Proxy.Enabled = true
+	authConf.SSH.Enabled = false
+	authConf.Proxy.DisableWebInterface = true
+	authConf.Proxy.DisableDatabaseProxy = true
+	require.NoError(auth.CreateEx(t, nil, authConf))
+	t.Cleanup(func() { require.NoError(auth.StopAll()) })
+
+	log.Info(">>> Start auth-proxy...")
+	require.NoError(auth.Start())
+
+	// ... and an SSH node connected via a reverse tunnel configured to
+	// reload after only a few failed connection attempts per minute
+	log.Info(">>> Creating and starting node...")
+	nodeCfg := service.MakeDefaultConfig()
+	nodeCfg.Hostname = Host
```

Comment:
Set DataDir here too? It expect it wouldn't matter for nodes, but not 100% sure.

Reply:
See `TeleInstance.startNode()` - `integration/helpers.go:706`